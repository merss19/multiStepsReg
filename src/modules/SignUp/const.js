"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Steps;
(function (Steps) {
    Steps[Steps["one"] = 1] = "one";
    Steps[Steps["two"] = 2] = "two";
    Steps[Steps["three"] = 3] = "three";
    Steps[Steps["four"] = 4] = "four";
})(Steps = exports.Steps || (exports.Steps = {}));
;
var Programs;
(function (Programs) {
    Programs["hero"] = "hero";
    Programs["mother"] = "mother";
    Programs["extreme"] = "extreme";
    Programs["tomorrow"] = "tomorrow";
})(Programs = exports.Programs || (exports.Programs = {}));
;
var ProgramsName;
(function (ProgramsName) {
    ProgramsName["hero"] = "#\u042F\u0433\u0435\u0440\u043E\u0439";
    ProgramsName["mother"] = "#\u041C\u0430\u043C\u0430 \u043C\u043E\u0436\u0435\u0442";
    ProgramsName["extreme"] = "#\u042D\u043A\u0441\u0442\u0440\u0438\u043C";
    ProgramsName["tomorrow"] = "#\u0417\u0430\u0432\u0442\u0440\u0430";
})(ProgramsName = exports.ProgramsName || (exports.ProgramsName = {}));
;
var ProgramNum;
(function (ProgramNum) {
    ProgramNum[ProgramNum["hero"] = 1] = "hero";
    ProgramNum[ProgramNum["mother"] = 2] = "mother";
    ProgramNum[ProgramNum["extreme"] = 3] = "extreme";
    ProgramNum[ProgramNum["tomorrow"] = 0] = "tomorrow";
    ProgramNum[ProgramNum["count"] = 4] = "count";
})(ProgramNum = exports.ProgramNum || (exports.ProgramNum = {}));
;
