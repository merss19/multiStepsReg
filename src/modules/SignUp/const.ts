export enum Steps {
  one = 1,
  two = 2,
  three = 3,
  four = 4
};

export enum Programs {
  hero = 'hero',
  mother = 'mother',
  extreme = 'extreme',
  tomorrow = 'tomorrow'
};

export enum ProgramsName {
  hero = '#Ягерой',
  mother = '#Мама может',
  extreme = '#Экстрим',
  tomorrow = '#Завтра'
};

export enum ProgramNum {
  hero = 1,
  mother = 2,
  extreme = 3,
  tomorrow = 0,
  count = 4
};