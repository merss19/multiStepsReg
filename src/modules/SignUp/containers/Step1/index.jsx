"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var main_1 = require("./main");
exports.StepOne = main_1.default;
var ducks_1 = require("./ducks");
exports.changeStep = ducks_1.changeStep;
exports.userCreate = ducks_1.userCreate;
exports.setProgramName = ducks_1.setProgramName;
exports.stepOneReducer = ducks_1.stepOneReducer;
var saga_1 = require("./saga");
exports.StepOneSaga = saga_1.StepOneSaga;
