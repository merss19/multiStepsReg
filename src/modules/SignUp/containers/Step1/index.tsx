import StepOne from './main';
import {changeStep, userCreate, setProgramName, stepOneReducer} from './ducks';
import {StepOneSaga} from './saga'

export {
  StepOne,
  changeStep, userCreate, stepOneReducer, setProgramName,
  StepOneSaga
};