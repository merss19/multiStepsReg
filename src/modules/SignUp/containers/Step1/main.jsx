"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var redux_form_1 = require("redux-form");
var react_router_dom_1 = require("react-router-dom");
var ducks_1 = require("./ducks");
var api_1 = require("./api");
var InputText_1 = require("components/InputText");
var RadioInput_1 = require("components/RadioInput");
var CheckboxInput_1 = require("components/CheckboxInput");
var Button_1 = require("components/Button");
var storage_1 = require("tools/storage");
var const_1 = require("../../const");
var CustomSteps_1 = require("../CustomSteps");
var CustomModal_1 = require("modules/CustomModal");
var ducks_2 = require("modules/Login/ducks");
var formValidations = require("tools/validations");
var ModalHOC_1 = require("components/HOC/ModalHOC");
var StepOneComponent = /** @class */ (function (_super) {
    __extends(StepOneComponent, _super);
    function StepOneComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    StepOneComponent.prototype.shouldComponentUpdate = function (nextProps) {
        return (nextProps.step !== this.props.step);
    };
    StepOneComponent.prototype.onSubmit = function (data) {
        var _this = this;
        this.props.userCreate(data, function (data) { return _this.resultAction(data); });
    };
    StepOneComponent.prototype.resultAction = function (data) {
        var _a = this.props, changeStep = _a.changeStep, setToken = _a.setToken;
        if (data && data.authToken) {
            setToken(data.authToken);
            storage_1.storage.save('token', data.authToken, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            storage_1.storage.save('promoName', data.promoName, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            changeStep(const_1.Steps.two);
        }
    };
    StepOneComponent.prototype.render = function () {
        var _this = this;
        var handleSubmit = this.props.handleSubmit;
        return (<div>
        {this.renderHeader()}
        <form className='entry__box' onSubmit={handleSubmit(function (data) { return _this.onSubmit(data); })}>
          <redux_form_1.Field name='email' placeholder='Ваш e-mail' type='text' component={InputText_1.default}/>
          <redux_form_1.Field name='password' placeholder='Ваш пароль' type='password' component={InputText_1.default}/>
          <redux_form_1.Field name='passwordAgain' placeholder='Пароль повторно' type='password' component={InputText_1.default}/>

          <div className='gender'>
            <div className='gender__group'>
              <p className='gender__title'>Ваш пол:</p>
              <redux_form_1.Field name='gender' id='male' value='male' type='radio' title='Мужчина' validate={formValidations.validateGender} component={RadioInput_1.default}/>
              <redux_form_1.Field name='gender' id='female' value='female' type='radio' title='Женщина' validate={formValidations.validateGender} component={RadioInput_1.default}/>
            </div>
            <hr className='gender__hr'/>
          </div>

          <div className='checkboxes'>
            <redux_form_1.Field name='accept' title='Принять условия ' id='accept' component={CheckboxInput_1.default}/>
          </div>

          <Button_1.Button type='submit' styleBtn={Button_1.ButtonTypes.info} wide={true}>
            Зарегистрироваться
          </Button_1.Button>


          <div className='entry__link text-center mtb20'>
            <div className='entry-nav__item'>
              <react_router_dom_1.Link to='/'>Войти</react_router_dom_1.Link>
            </div>
            <div className='entry-nav__item'>
              <react_router_dom_1.Link to='/restore'>Забыли пароль?</react_router_dom_1.Link>
            </div>
          </div>
        </form>
      </div>);
    };
    return StepOneComponent;
}(CustomSteps_1.default));
exports.StepOneComponent = StepOneComponent;
exports.StepOne = react_redux_1.connect(null, {
    toggleModal: CustomModal_1.toggleModal,
    setError: ducks_1.setError,
    userCreate: ducks_1.userCreate,
    setToken: ducks_2.setToken
})(StepOneComponent);
exports.default = ModalHOC_1.modalHOC(redux_form_1.reduxForm({
    form: 'StepOne',
    validate: formValidations.validateSignUp,
    asyncValidate: api_1.asyncValidate,
    asyncBlurFields: ['email']
})(exports.StepOne), ducks_1.moduleName, ducks_1.subModuleName);
