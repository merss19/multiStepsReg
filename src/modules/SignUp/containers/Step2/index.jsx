"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var main_1 = require("./main");
exports.StepTwo = main_1.default;
var ducks_1 = require("./ducks");
exports.stepTwoReducer = ducks_1.stepTwoReducer;
var saga_1 = require("./saga");
exports.StepTwoSaga = saga_1.StepTwoSaga;
