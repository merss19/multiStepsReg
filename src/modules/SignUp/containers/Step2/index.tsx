import StepTwo from './main';
import {stepTwoReducer} from './ducks';
import {StepTwoSaga} from './saga'


export {
  StepTwo,
  stepTwoReducer,
  StepTwoSaga
};
