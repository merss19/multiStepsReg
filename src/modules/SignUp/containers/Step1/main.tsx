import * as React from 'react';
import {connect} from 'react-redux';
import {Field, FormProps, reduxForm} from 'redux-form';
import {Link} from 'react-router-dom';
import {setError, userCreate, moduleName, subModuleName} from './ducks';
import {asyncValidate} from './api';
import InputText from 'components/InputText';
import RadioInput from 'components/RadioInput';
import CheckboxInput from 'components/CheckboxInput';
import {SignUpFormData} from '../../interfaces';
import {Button, ButtonTypes} from 'components/Button';
import {storage} from 'tools/storage';
import {Steps} from '../../const';
import CustomSteps, {CustomStepsProps} from '../CustomSteps';
import {toggleModal} from "modules/CustomModal";
import {setToken} from 'modules/Login/ducks'
import * as formValidations from 'tools/validations'
import {modalHOC} from "components/HOC/ModalHOC";

interface OwnProps extends CustomStepsProps, FormProps<{}, {}, {}> {
  changeStep: (data: Steps) => void
  step: Steps
}

type ConnectedState = {
  isFetching: boolean
  isError: boolean
  errMsg: string
}
type ConnectedDispatch = {
  getPrograms(): void
  setToken(token: string): void
  toggleModal(data: boolean): void
  setError(data: boolean): any
  userCreate(data: SignUpFormData, cb: (data: any) => void): any
}

type Props = OwnProps & ConnectedDispatch;

export class StepOneComponent extends CustomSteps<Props> {

  shouldComponentUpdate(nextProps: Props) {
    return (
      nextProps.step !== this.props.step
    );
  }

  onSubmit(data: SignUpFormData) {
    this.props.userCreate(data, (data) => this.resultAction(data));
  }

  resultAction(data: any): void {
    const {changeStep, setToken} = this.props;
    if (data && data.authToken) {
      setToken(data.authToken);
      storage.save('token', data.authToken,
        {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      storage.save('promoName', data.promoName,
        {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      changeStep(Steps.two);
    }
  }

  render() {
    const {handleSubmit} = this.props;
    return (
      <div>
        {this.renderHeader()}
        <form className='entry__box' onSubmit={handleSubmit((data: any) => this.onSubmit(data))}>
          <Field
            name='email'
            placeholder='Ваш e-mail'
            type='text'
            component={InputText}
          />
          <Field
            name='password'
            placeholder='Ваш пароль'
            type='password'
            component={InputText}
          />
          <Field
            name='passwordAgain'
            placeholder='Пароль повторно'
            type='password'
            component={InputText}
          />

          <div className='gender'>
            <div className='gender__group'>
              <p className='gender__title'>Ваш пол:</p>
              <Field
                name='gender'
                id='male'
                value='male'
                type='radio'
                title='Мужчина'
                validate={formValidations.validateGender}
                component={RadioInput}
              />
              <Field
                name='gender'
                id='female'
                value='female'
                type='radio'
                title='Женщина'
                validate={formValidations.validateGender}
                component={RadioInput}
              />
            </div>
            <hr className='gender__hr'/>
          </div>

          <div className='checkboxes'>
            <Field
              name='accept'
              title='Принять условия '
              id='accept'
              component={CheckboxInput}/>
          </div>

          <Button
            type='submit'
            styleBtn={ButtonTypes.info}
            wide={true}
          >
            Зарегистрироваться
          </Button>


          <div className='entry__link text-center mtb20'>
            <div className='entry-nav__item'>
              <Link to='/'>Войти</Link>
            </div>
            <div className='entry-nav__item'>
              <Link to='/restore'>Забыли пароль?</Link>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

export const StepOne = connect(null, {
  toggleModal,
  setError,
  userCreate,
  setToken
})(StepOneComponent);

export default modalHOC(reduxForm({
  form: 'StepOne',
  validate: formValidations.validateSignUp,
  asyncValidate,
  asyncBlurFields: ['email']
})(StepOne), moduleName, subModuleName);