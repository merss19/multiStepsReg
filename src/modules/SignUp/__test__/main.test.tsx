import {shallow, mount} from 'enzyme';
import {SignUpComponent, SignUp} from '../main';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import * as sinon from 'sinon';
import {Provider} from 'react-redux';
import configureStore from '../../../store';
import {BrowserRouter as Router} from 'react-router-dom';
import {CustomModal, ModalTypes} from '../../CustomModal';
import TodayTask from '../../../pages/PageTodayTask';
import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {storage} from '../../../tools/storage/index';
import {Steps} from '../const';
import {queryParse} from '../../../tools/utils/index';
import {StepOne} from '../containers/Step1';
import {StepTwo} from '../containers/Step2';
import {StepThree} from '../containers/Step3';
import {StepFour} from '../containers/Step4';
import LogoLink from "../../../components/LogoLink/index";
import * as Immutable from 'immutable';

//const store = configureStore();
configure({adapter: new Adapter()});
let changeStep,
  getPrograms,
  userCreate,
  setProgramName,
  setToken,
  isFetching,
  isLoad,
  isError,
  step,
  match,
  location,
  history;
const start = () => {
  beforeEach(() => {
    changeStep = jest.fn(),
      getPrograms = jest.fn(),
      userCreate = jest.fn(),
      setProgramName = jest.fn(),
      setToken = jest.fn(),
      isFetching = false,
      isLoad = false,
      isError = false,
      step = Steps.one,
      match = {
        isExact: true,
        params: {},
        path: "/task",
        url: "/task"
      },
      location = {},
      history = {
        replace: jest.fn()
      }
  });
  const props = {
    getPrograms: getPrograms,
    userCreate: userCreate,
    changeStep: changeStep,
    setProgramName: setProgramName,
    setToken: setToken,
    step: step,
    isFetching: isFetching,
    isLoad: isLoad,
    isError: isError,
    match: match,
    location: location,
    history: history
  };
  const buildComponent = () => {
    return shallow(<SignUpComponent {...props}/>)
  };
  return {
    props,
    buildComponent
  }
};
describe('Component SignUp', () => {
  let component;
  start();
  beforeEach(() => {
    component = start().buildComponent();
  });
  it('renders component', () => {
    expect(component).toHaveLength(1);
  });
  /*  it('renders component Mount', () => {
      const component = start().buildComponentMount();
      expect(component).toHaveLength(1);
    });*/
  it('renders correctly', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
  describe('Test props', () => {
    it('pass prop program', () => {
      expect(component.instance().props.changeStep).toBe(start().props.changeStep);
      expect(component.instance().props.userCreate).toBe(start().props.userCreate);
      expect(component.instance().props.setProgramName).toBe(start().props.setProgramName);
    });
  });
  describe('Test componentWillMount', () => {
    it('componentWillMount, token=test', () => {
      location = {
        search: '?token=test'
      };
      component = start().buildComponent();
      expect(changeStep).not.toHaveBeenCalled();
      expect(setProgramName).not.toHaveBeenCalled();
    });
    it('componentWillMount, step=1', () => {
      location = {
        search: '?step=1'
      };
      component = start().buildComponent();
      let query: any = queryParse(component.instance().props.location.search);
      expect(changeStep).toHaveBeenCalledWith(parseInt(query.step));
      expect(setProgramName).not.toHaveBeenCalled();
    });
    it('componentWillMount, step=5', () => {
      location = {
        search: '?step=5'
      };
      component = start().buildComponent();
      let query: any = queryParse(component.instance().props.location.search);
      expect(changeStep).not.toHaveBeenCalled();
      expect(setProgramName).not.toHaveBeenCalled();
    });
    it('componentWillMount, name=hero', () => {
      location = {
        search: '?name=hero'
      };
      component = start().buildComponent();
      let query: any = queryParse(component.instance().props.location.search);
      expect(changeStep).not.toHaveBeenCalled();
      expect(setProgramName).toHaveBeenCalledWith(query.name);
    });
    it('componentWillMount, name=hero2', () => {
      location = {
        search: '?name=hero2'
      };
      component = start().buildComponent();
      let query: any = queryParse(component.instance().props.location.search);
      expect(changeStep).not.toHaveBeenCalled();
      expect(setProgramName).not.toHaveBeenCalled();
    });
    it('componentWillMount, step=1&name=hero', () => {
      location = {
        search: '?step=1&name=hero'
      };
      component = start().buildComponent();
      let query: any = queryParse(component.instance().props.location.search);
      expect(changeStep).toHaveBeenCalledWith(parseInt(query.step));
      expect(setProgramName).toHaveBeenCalledWith(query.name);
    });
  });
  it('Test shouldComponentUpdate', () => {
    const instance = component.instance() as SignUpComponent;
    step = Steps.two;
    let props = {
      ...start().props,
      step:step
    };
    const shouldUpdate = instance.shouldComponentUpdate(props);
    expect(shouldUpdate).toBe(true)
  });
  describe('Test renderContent', () => {
    it('Steps.one', () => {
      expect(component.find(StepOne).exists()).toBe(true);
      expect(component.find(StepTwo).exists()).toBe(false);
      expect(component.find(StepThree).exists()).toBe(false);
      expect(component.find(StepFour).exists()).toBe(false);
    });
    it('Steps.two', () => {
      step = Steps.two;
      component.setProps({step: step});
      expect(component.find(StepOne).exists()).toBe(false);
      expect(component.find(StepTwo).exists()).toBe(true);
      expect(component.find(StepThree).exists()).toBe(false);
      expect(component.find(StepFour).exists()).toBe(false);
    });
    it('Steps.three', () => {
      step = Steps.three;
      component.setProps({step: step});
      expect(component.find(StepOne).exists()).toBe(false);
      expect(component.find(StepTwo).exists()).toBe(false);
      expect(component.find(StepThree).exists()).toBe(true);
      expect(component.find(StepFour).exists()).toBe(false);
    });
    it('Steps.four', () => {
      step = Steps.four;
      component.setProps({step: step});
      expect(component.find(StepOne).exists()).toBe(false);
      expect(component.find(StepTwo).exists()).toBe(false);
      expect(component.find(StepThree).exists()).toBe(false);
      expect(component.find(StepFour).exists()).toBe(true);
      step = 5;
      component.setProps({step: step});
      expect(component.find(StepOne).exists()).toBe(false);
      expect(component.find(StepTwo).exists()).toBe(false);
      expect(component.find(StepThree).exists()).toBe(false);
      expect(component.find(StepFour).exists()).toBe(true);
    });
  });
  describe('Test render()', () => {
    it('render LogoLink', () => {
      expect(component.find(LogoLink).exists()).toBe(true);
    });
    it('pass prop step', () => {
      expect(component.instance().props.step).toBe(start().props.step);
      expect(component.find('.entry-step__no').text()).toBe('#' + start().props.step);
    });
    it('call renderContent', () => {
      const spy = sinon.spy(SignUpComponent.prototype, 'renderContent');
      component = start().buildComponent();
      expect(spy.called).toBe(true);
    });
  });
});


