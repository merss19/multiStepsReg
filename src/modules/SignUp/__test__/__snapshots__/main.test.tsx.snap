// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SignUp renders correctly 1`] = `
<div
  className="layout layout--entry"
>
  <div
    className="grid entry-header"
  >
    <div
      className="1/4--desk grid__cell todayme-logo"
    >
      <LogoLink />
    </div>
    <div
      className="2/4--desk grid__cell text-center"
    >
      <div
        className="entry-step"
      >
        <span
          className="entry-step__title"
        >
          Шаг
        </span>
        <span
          className="entry-step__no"
        >
          #
          1
        </span>
        <span
          className="entry-step__go-to"
        >
          из 4
        </span>
      </div>
    </div>
  </div>
  <div
    className="entry entry--min"
  >
    <div
      className="entry__inner"
    >
      <Connect(withModal(ReduxForm))
        changeStep={[Function]}
        getPrograms={[Function]}
        history={
          Object {
            "replace": [Function],
          }
        }
        isError={false}
        isFetching={false}
        isLoad={false}
        location={Object {}}
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/task",
            "url": "/task",
          }
        }
        setProgramName={[Function]}
        setToken={[Function]}
        step={1}
        userCreate={[Function]}
      />
    </div>
  </div>
</div>
`;
